app/helpers/random_assign_helper.rb

```diff
@@ -56,7 +59,8 @@ def initialize_data_structures(num_groups, reviewer_ids, reviewee_ids)
   #
   # Also add existing peer reviews to @reviewer_to_reviewee_sets.
   def process_existing_peer_reviews(reviewer_ids)
-    PeerReview.includes(:reviewee)
+    PeerReview.includes(:reviewer,
+                        { result: { submission: :grouping } })
```

Comment:
This is the only part of this commit that is not including the other changes (I forked off of the PR fix so I could test this stuff worked).